staging/src/k8s.io/component-base/logs/api/v1/options.go

```diff
@@ -50,58 +61,186 @@ func NewOptions() *Options {
 //
 // The optional FeatureGate controls logging features. If nil, the default for
 // these features is used.
-func (o *Options) ValidateAndApply(featureGate featuregate.FeatureGate) error {
-	errs := o.validate()
+func (c *LoggingConfiguration) ValidateAndApply(featureGate featuregate.FeatureGate) error {
```

Comment:
Moving the pflag support into a wrapper type makes sense to me - done and squashed. I kept the helper code exported just in case that it might be needed elsewhere, but right now it is only used internally and thus could also be removed from the API of the package.

I'm less sure about the config functions. It makes the type inconsistent with other, similar types. The godoc documentation becomes harder to read. Callers need an additional import. For example:
```patch
diff --git a/cmd/kube-controller-manager/app/options/options.go b/cmd/kube-controller-manager/app/options/options.go
index 5c7a15b4aac..abe4e4e533b 100644
--- a/cmd/kube-controller-manager/app/options/options.go
+++ b/cmd/kube-controller-manager/app/options/options.go
@@ -35,6 +35,7 @@ import (
        cpoptions "k8s.io/cloud-provider/options"
        cliflag "k8s.io/component-base/cli/flag"
        "k8s.io/component-base/logs"
+       logsapi "k8s.io/component-base/logs/api/v1"
        "k8s.io/component-base/metrics"
        cmoptions "k8s.io/controller-manager/options"
        kubectrlmgrconfigv1alpha1 "k8s.io/kube-controller-manager/config/v1alpha1"
@@ -257,7 +258,7 @@ func (s *KubeControllerManagerOptions) Flags(allControllers []string, disabledBy
        s.SAController.AddFlags(fss.FlagSet("serviceaccount controller"))
        s.TTLAfterFinishedController.AddFlags(fss.FlagSet("ttl-after-finished controller"))
        s.Metrics.AddFlags(fss.FlagSet("metrics"))
-       s.Logs.AddFlags(fss.FlagSet("logs"))
+       logsapi.AddFlags(s.Logs, fss.FlagSet("logs"))
 
        fs := fss.FlagSet("misc")
        fs.StringVar(&s.Master, "master", s.Master, "The address of the Kubernetes API server (overrides any value in kubeconfig).")
```

I've also changed those, but kep that change in a separate commit in case that we want to back it out again.

Reply:
> It makes the type inconsistent with other, similar types

I don't see Validation / Defaulting / AddFlags methods defined on any other API types. The distinct functions lgtm

Reply:
I was thinking of types like the ones behind s.Metrics, s.TTLAfterFinishedController, etc. above, like [component-base/metrics.Options](https://github.com/kubernetes/kubernetes/blob/19c8a2127177b43effb9bfe1de272d6f08ea56e8/staging/src/k8s.io/component-base/metrics/options.go#L29-L97). Those are not API types (yet), but perhaps should be.

Anyway, I'm fine with distinct functions, so let's use that.